File: json_memory_tests/ruby/nested_50.rb
require 'json'
require 'time'

parent = { "val" => 0, "child" => nil }
t0 = Time.now
for i in 1..50
  parent = { "val" => i, "child" => parent }
end

serialized = JSON.generate(parent)
parsed = JSON.parse(serialized)

explorer = parsed
for k in 1..25
  explorer = explorer["child"]
end
val_check = explorer["val"]

t1 = Time.now
json_ms = (t1 - t0) * 1000.0
puts "json_elapsed_ms: #{'%.3f' % json_ms}"

File: json_memory_tests/node/nested_50.js
let parent = { val: 0, child: null };

const t0 = performance.now();

for (let i = 1; i <= 50; i++) {
    parent = { val: i, child: parent };
}

const serialized = JSON.stringify(parent);
const parsed = JSON.parse(serialized);

let explorer = parsed;
for (let k = 1; k <= 25; k++) {
    explorer = explorer.child;
}
const val_check = explorer.val;

const t1 = performance.now();
const json_ms = t1 - t0;
console.log("json_elapsed_ms: " + json_ms.toFixed(3));
